extensions/pkg/controller/worker/genericactuator/machine_controller_manager.go

```diff
@@ -84,7 +92,12 @@ func (a *genericActuator) deployMachineControllerManager(ctx context.Context, lo
 		return fmt.Errorf("waiting until deployment/%s is updated: %w", McmDeploymentName, err)
 	}
 
-	return nil
+	// TODO(rfranzke/BeckerMax): Remove in a future release.
+	secretName := a.mcmName
+	if !a.useTokenRequestor {
+		secretName = "shoot-access-" + a.mcmName
+	}
+	return kutil.DeleteObject(ctx, a.client, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: secretName, Namespace: workerObj.Namespace}})
```

Comment:
Do we also need to delete secrets when deleteMachineControllerManager is called?

Reply:
The code for deletion of the secrets was missing already before (error-pronely assuming that the `Worker` object is only deleted when the `Namespace` itself will also be deleted soon afterwards, i.e., relying on the namespace cleanup). Good catch, I've added it now.